lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2053,6 +2068,9 @@ SILLocation::DebugLoc IRGenDebugInfoImpl::decodeSourceLoc(SourceLoc SL) {
   auto &Cached = DebugLocCache[SL.getOpaquePointerValue()];
   if (Cached.Filename.empty())
     Cached = SILLocation::decode(SL, SM);
+  if (Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView)
```

Comment:
Is it a requirement to strip the columns in order to not confuse WinDbg or could you just leave them in?

Reply:
WinDbg does technically work with column information but the user must select an address for every line with multiple columns, which makes it pretty annoying. Also clang does not emit column information for C++.

Reply:
It would be great if we could limit this set-column-to-zero code to as few places as possible, so we don't forget to do it when writing new code. This entire code looks like it is redundant with IRGenDebugInfoImpl::decodeSourceLoc() so perhaps we could use that function here and sink the check into it?